redbot/cogs/mod/settings.py

```diff
@@ -87,35 +99,131 @@ async def hierarchy(self, ctx: commands.Context):
                 _("Role hierarchy will be ignored when moderation commands are issued.")
             )
 
-    @modset.command()
+    @modset.group()
+    @commands.guild_only()
+    async def mentionspam(self, ctx: commands.Context):
+        """
+        Manage the automoderation settings for mentionspam
+        """
+
+    @mentionspam.command(name="warn")
+    @commands.guild_only()
+    async def mentionspam_warn(self, ctx: commands.Context, max_mentions: int = 0):
+        """
+        Sets the autowarn conditions for mention spam.
+
+        Users will be warned if they send any messages which contain more than
+        `<max_mentions>` mentions.
+
+        `<max_mentions>` must be at least 1. Set to 0 to disable.
+        """
+        mention_spam = await self.config.guild(ctx.guild).mention_spam.all()
+        if not max_mentions:
+            if not mention_spam["warn"]:
+                return await ctx.send(_("Autowarn for mention spam is already disabled."))
+            await self.config.guild(ctx.guild).mention_spam.warn.set(False)
+            return await ctx.send(_("Autowarn for mention spam disabled."))
+
+        if max_mentions < 1:
+            return await ctx.send(_("`<max_mentions>` must be 1 or higher to autowarn."))
```

Comment:
This entire section looks confusing to me. The docstring says "**Set** to 0 to disable" (never stating that is the default value of the command), the `if not max_mentions` part looks like it is accepting `0` to view the current settings (consider using `None` and testing `if max_mentions is None`, and any value below `1` (including 0) is blocked.

Look through this part of the code again, and consider what it should actually be *doing* for every possible value.

(This comment applies to all 3 subcommands of `[p]mentionspam`)

Reply:
Don't know a better way to word it either, though it still gets the point across regardless. If someone has a better wording for this, I'm happy to change. 

Added information that it defaults if no value is passed.

Reply:
You could take a page from [p]set deletedelay. Still not the best, but at least it'd be consistent.
Must be 0 or greater. Set to 0 to disable this feature.

Reply:
Possibly, I'll look into adding this change when I can later today and see. Thanks for this! :).

Reply:
Done, thanks @zephyrkul. Changes looked a lot cleaner so decided to go with that for the mention spam.